models/note.go

```diff
@@ -22,16 +23,33 @@ var categoryStrings = [...]string{
 	"predictions",
 }
 
-func (category Category) String() string {
-	if category < MARGINALIA || category > PREDICTIONS {
-		return "Unknown"
+var UnDeserializeableCategoryStringError = errors.New("String does not correspond to a Note Category")
+
+func DeserializeCategory(input string) (Category, error) {
+	for i := 0; i < len(categoryStrings); i++ {
+		if input == categoryStrings[i] {
+			return Category(i), nil
+		}
 	}
+	return MARGINALIA, UnDeserializeableCategoryStringError
+}
 
+func (category Category) String() string {
 	return categoryStrings[category]
 }
 
 type Note struct {
+	Id           int64     `json:"id"`
```

Comment:
why are we adding id to note?

Reply:
In the ui it'll be important to know the note id, so we can update and delete properly. The only time we won't know the id is when its new and hasn't been saved.